src/main/java/fr/insalyon/creatis/gasw/plugin/listener/healing/execution/CommandState.java

```diff
@@ -308,7 +320,103 @@ public void addUploadTime(long time) {
+            if (this.jobErrorRate >= HealingConfiguration.getInstance().getMaxErrorJobPercentage() ||
+                    this.invocationPartialErrorRate >= HealingConfiguration.getInstance().getMaxErrorInvocationPercentage()) {
+                this.killAllJobs = true;
+                logger.info("[Healing] Attention, updating killing decision to true. Nm min invocations are "+
+                        HealingConfiguration.getInstance().getMinInvocations()+" , job error rate is "+
+                        this.jobErrorRate + " and invocation error rate is "+ this.invocationPartialErrorRate);
+            }
+        }
+
+    }
+
+    private void killAllJobs() {
+        logger.info("[Healing] Killing all jobs of type " + this.command);
+        try {
+            JobDAO jobDAO = DAOFactory.getDAOFactory().getJobDAO();
+            List<Integer> invocationIDs = jobDAO.getInvocationsByCommand(this.command);
+            for (int invocation : invocationIDs) {
+                killInvocationJobs(invocation);
+            }
+
+        } catch (DAOException ex) {
+            logger.error("[Healing] Error killing jobs: ", ex);
+        }
+    }
+
+    private void killInvocationJobs (int invocation) {
+        logger.info("[Healing] Killing jobs of invocation " + invocation);
+        try {
+            JobDAO jobDAO = DAOFactory.getDAOFactory().getJobDAO();
+            GaswStatus status = GaswStatus.KILL;
+            List<Job> activeJobs = jobDAO.getActiveJobsByInvocationID(invocation);
+            if ((activeJobs != null) && (!activeJobs.isEmpty())) {
+                for (Job job : activeJobs) {
+                    job.setStatus(status);
+                    jobDAO.update(job);
+                    logger.info("[Healing] Setting status of job " + job.getId() + " to " + status);
+                    //all subsequent jobs are replica, so kill them as such
+                    status = GaswStatus.KILL_REPLICA;
+                }
+            } else {
+                if (jobDAO.getNumberOfCompletedJobsByInvocationID(invocation) == 0) {
+                    handleHeldJobs(invocation);
+                }
+            }
+
+        } catch (DAOException ex) {
+            logger.error("[Healing] Error killing jobs for invocation : " + invocation, ex);
+        }
+    }
+
+    private void handleHeldJobs (int invocationID){
+        logger.info("[Healing] Handle Held jobs for invocation " + invocationID);
+        try {
+            JobDAO jobDAO = DAOFactory.getDAOFactory().getJobDAO();
+            List<Job> failedJobs = jobDAO.getFailedJobsByInvocationID(invocationID);
+            if ((failedJobs!=null) && (!failedJobs.isEmpty())) {
+                for (Job job : failedJobs) {
+                    if ((job.getStatus() == GaswStatus.ERROR_HELD) || (job.getStatus() == GaswStatus.STALLED_HELD)) {
```

Comment:
This will be done again at each healing cycle and I'not sure it is good to signal several time to moteur that the same invocation is finished.
Maybe we could update their status to FAILED/STALLED as it wont be possible to unhold them anyway.

Reply:
Right. I propose we set the status to Cancelled.

Reply:
Status updated to Failed/Stalled